Feature/1 update arguments (#7)
semanticarts/undefined-concepts-tool: README.md

# undefined-concepts-tool

> Find all of the undefined concepts in an ontology

Reviewer: ```suggestion
Find concepts that are referenced but undefined in an ontology.
```
@dylan-sa your summary of the function is better.

## Installation

> The script is tested with python 3.10. But most later versions of python3 should work.

Reviewer 2: Your setuptools descriptor can explicitly document the Python version dependency.

The script requires the `rdflib` library. Install with `pip install rdflib`.

> TODO: This should be either created as a package, or a virtual environment to reduce the

Reviewer 3: Agreed, and you should have a requirements.txt file to state the dependencies.

Reviewer: I almost added it to this PR, but if we are going to move this into a different repo, or expand the scope of the repo, it would be better to wait.

Reviewer 2: `requirements.txt` is outdated for packages intended for distribution, [setuptools](https://setuptools.pypa.io/en/latest/) is the preferred route.

errors that might crop up in dependencies.

## Usage

> The command takes to arguments context and a focus graph. The context should include all of the

Reviewer 3: ```suggestion
The command takes two arguments context and a focus graph. The context should include all of the
```

ontology files that are imported by the focus graph. These files can all be in a directory, or specified
individually.
For example imagine a structure:

```
project
|--ontologies
|  |--- ontologyA.ttl
|  |--- ontolgoyB.ttl
|--taxonomies
|  |--- taxonomyA.ttl
|  |--- bad_taxonomy.ttl
|-- focus_ontology_a.ttl
|--- focus_ontology_b.ttl
```

> If one does not want to include the file `bad_taxonomy.ttl`, run `./find_undefined_concepts focus_ontology_a.ttl focus_ontology_b.ttl -c ontologies/ taxonomies/taxonomyA.ttl`.

Reviewer 3: I find this complex to the point of incomprehensible. Why not keep it simple? It takes a series of arguments of files and/or folders. It will operate on each individual file, and all the files in a folder. We could save for later the idea of skipping files in a folder, but I think that's an advanced feature that isn't needed at this stage.

Reviewer: The idea is you can add directories or individual files. Its not really skipping a file, its just allowing one to pass exactly what they want.

Reviewer 3: OK, I must have misunderstood what you are doing. That is the required functionality.

> Or one could run: `./find_undefined_concepts -c ontologies/ taxonomies/taxonomyA.ttl -- focus_ontology_a.ttl focus_ontology_b.ttl`.

Reviewer 3: Same comment here as above.

> Everything after the `-c` or `--context` flag is treated as an array of either files or directories. The focus graph file or files needs to appear before the flag, or after a `--`, or directly after the command if there is no context flag added.

Reviewer 3: Same here. Note that the files where you are looking for references and the files where you are looking for definitions make up the same set of files. There is no focus other than those files.

Reviewer: There is though. We aren't looking for all undefined concepts in the large ontology project, just the ones that are in the specific domain ontology we are currently working with. At least, that is the scope of the current code.

Reviewer 3: Hm, OK, that's not what I thought it was doing.
